### PR TITLE
The Witness: Rename some *horrendously* named variables

### PR DESCRIPTION
**Pull request by NewSoupVi**
Also reorder them slightly

Basically, Witness code refers to "Progression" as "Prog". But then it introduced progressive items, so in order to avoid confusion, I called that "Multi" instead.

This just makes everything explicit
"Multi" -> "Progressive"
"No Multi" -> "Base" (Meaning: Not Progressive)
"Prog" -> "Progression"

Split off from https://github.com/ArchipelagoMW/Archipelago/pull/3961 in an effort to make that PR more reviewable afterwards

Tested:
Nah, I trust PyCharms Refactor->Rename

### CODE REVIEW

File: worlds/witness/player_logic.py [hunk 1]
```diff
@@ -868,18 +870,18 @@ def finalize_items(self) -> None:
...
         """
-        for item in self.PROG_ITEMS_ACTUALLY_IN_THE_GAME_NO_MULTI:
+        for item in self.BASE_PROGESSION_ITEMS_ACTUALLY_IN_THE_GAME:
             if item not in self.THEORETICAL_ITEMS:
                 progressive_item_name = static_witness_logic.get_parent_progressive_item(item)
-                self.PROG_ITEMS_ACTUALLY_IN_THE_GAME.add(progressive_item_name)
+                self.PROGRESSION_ITEMS_ACTUALLY_IN_THE_GAME.add(progressive_item_name)
                 child_items = cast(ProgressiveItemDefinition,
                                    static_witness_logic.ALL_ITEMS[progressive_item_name]).child_item_names
                 multi_list = [child_item for child_item in child_items
```
Comment (Exempt-Medic) [inline worlds/witness/player_logic.py:879]: `multi_list` ?

File: worlds/witness/rules.py [hunk 2]
```diff
@@ -202,7 +202,7 @@ def _has_item(item: str, world: "WitnessWorld",
         return lambda state: _can_do_theater_to_tunnels(state, world)
 
     prog_item = static_witness_logic.get_parent_progressive_item(item)
```
Comment (Exempt-Medic) [inline worlds/witness/rules.py:204]: `prog_item` ?

**Comment by Exempt-Medic** [comment]
`hints.py` has the variables `prog_amount`, `prog_item_names_in_this_world`, and `prog_items_in_this_world`

**Comment by NewSoupVi** [comment]
> `hints.py` has the variables `prog_amount`, `prog_item_names_in_this_world`, and `prog_items_in_this_world`

Done as well

**Review by Exempt-Medic — APPROVED** [review]
All renaming look correct to me

**Review by Berserker66 — APPROVED** [review]

**Review by nicholassaylor — APPROVED** [review]
LGTM

**Pull request merged**